File: Lecture-6/Lecture/Code/PriorityQueueItem.py
class PriorityQueueItem:

    """Provides a handle for a queue item.

    This handle can be used to remove or
    update the queue item.
    """
    
    def __init__(self, key, value, index):
        self.key = key;
        self.value = value;
        self.index = index;

File: Lecture-6/Lecture/Code/PriorityQueueItem.java
/**
 * Provides a handle for an inserted queue
 * item. This handle can be used to remove
 * or update the queue item.
 *
 * @param <T> The type of the data
 */
public class PriorityQueueItem<T> {
	// These member variables are
	// package-private
	int key;
	T value;
	int heapIndex;
}